packages/flutter/lib/src/foundation/serialization.dart

```diff
@@ -30,33 +30,33 @@ class WriteBuffer {
   }
 
   /// Write a Uint16 into the buffer.
-  void putUint16(int value) {
-    _eightBytes.setUint16(0, value, Endian.host);
+  void putUint16(int value, [Endian endian]) {
```

Comment:
instead of position arguement, we can use keyword argument with default value = Endian.host
```void putUint16(int value, {Endian endian = Endian.host})```

Reply:
Will the endian ever change for a WriteBuffer? if it will not, we can make it a constructor parameter

Reply:
> instead of position arguement, we can use keyword argument with default value = Endian.host
> `void putUint16(int value, {Endian endian = Endian.host})`

OK. I'll update the PR

Reply:
> Will the endian ever change for a WriteBuffer? if it will not, we can make it a constructor parameter

Yeah, some case need different endian with one WriteBuffer/ReadBuffer. And that is how the JAVA `ByteBuffer` with `ByteOrder` works

```java
ByteBuffer buffer = ByteBuffer.allocate(Integer.SIZE * 2);
buffer.order(ByteOrder.LITTLE_ENDIAN).putInt(1);
buffer.order(ByteOrder.BIG_ENDIAN).putInt(1);
```

Reply:
> instead of position arguement, we can use keyword argument with default value = Endian.host
> `void putUint16(int value, {Endian endian = Endian.host})`

@chunhtai  I've tried with `Endian.host` but ended with error "The default value of an optional parameter must be constant.dart(non_constant_default_value)"

<img width="992" alt="Screen Shot 2020-01-07 at 6 36 35 AM" src="https://user-images.githubusercontent.com/7928961/71854263-8e9f4480-3118-11ea-8bf6-eb11a84d5c37.png">

Reply:
This is fine, I wasn't aware Endian.host is a const.